Sources/AsyncStream.swift

```diff
@@ -1,7 +1,7 @@
 public protocol AsyncStream {
     var closed: Bool { get }
     func close() -> Bool
-    func receive(length length: Int, result: (Void throws -> Data) -> Void)
-    func send(data data: Data, result: (Void throws -> Void) -> Void)
+    func receive(upTo byteCount: Int, result: (Void throws -> Data) -> Void)
```

Comment:
.receive(upTo: 1024) { in 
  return Data()
}

When read says "receive up to 1024 in the execution of" and I ask "1024 of what?"
Could we some how rephrase this to read as a literal statement that is self constant? Maybe adding a extensions to Int for bytes, kilobytes, megabytes, etc that each convert the Int into it's byte equivalent so that we can write it as:

.receive(upTo: 1.kilobyte) { in 
  return Data()
}